### PR TITLE
Multi-buy discount for custom line items

### LINKED ISSUE #1593: Multi-buy discount for custom line items
Our SDK should add support for the new Multi-buy discount for custom line items.

### PR DESCRIPTION
**Pull request by acbeni**
Close #1593

### CODE REVIEW

File: commercetools-internal-docs/src/main/java/io/sphere/sdk/meta/ReleaseNotes.java [hunk 1]
```diff
@@ -133,6 +133,13 @@
  <li class=fixed-in-release></li>
  </ul>
  -->
+ <h3 class=released-version id="v1_25_0">1.26.0 (19.10.2017)</h3>
```
Comment (katmatt) [inline commercetools-internal-docs/src/main/java/io/sphere/sdk/meta/ReleaseNotes.java]: I like your idea to add release notes on the branch 👍 But we should put the correct version into the id attribute `id="v1_26_0"` 😉

File: commercetools-models/src/main/java/io/sphere/sdk/cartdiscounts/SelectionMode.java [hunk 2]
```diff
@@ -4,7 +4,7 @@
 import io.sphere.sdk.models.SphereEnumeration;
 
 /**
- * Determines which of the items, participating in a {@link MultiBuyLineItemsTarget},
+ * Determines which of the items, participating in a {@link MultiBuyLineItemsTarget} and a {@link MultiBuyCustomLineItemsTarget},
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/cartdiscounts/SelectionMode.java]: Nice that you added the new target type here too 🥇

File: commercetools-models/src/main/java/io/sphere/sdk/cartdiscounts/MultiBuyCustomLineItemsTarget.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package io.sphere.sdk.cartdiscounts;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import io.sphere.sdk.models.Base;
+
+import javax.annotation.Nullable;
+
+/**
+ * The multi-buy discount represents a "Buy 4 items, get 2 of them at a discounted rate" type of discount. This discount target is very similar
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/cartdiscounts/MultiBuyCustomLineItemsTarget.java]: I like this javadoc 👍

File: commercetools-models/src/main/java/io/sphere/sdk/cartdiscounts/AbstractMultiBuyTarget.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
+package io.sphere.sdk.cartdiscounts;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import io.sphere.sdk.models.Base;
+
+import javax.annotation.Nullable;
+
+
+abstract class AbstractMultiBuyTarget extends Base implements CartDiscountTarget {
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/cartdiscounts/AbstractMultiBuyTarget.java]: Good idea to extract the commonalities in this abstract class.

File: commercetools-internal-docs/src/main/java/io/sphere/sdk/meta/ReleaseNotes.java (hunk 1 above)
Comment (acbeni) [reply commercetools-internal-docs/src/main/java/io/sphere/sdk/meta/ReleaseNotes.java]: 😅 yeah this is a stupid mistake

**Comment by acbeni** [comment]
@katmatt now its all good i guess 😄 , WDYT?

**Pull request merged**